monai/transforms/spatial/array.py

```diff
@@ -116,20 +112,28 @@ def __call__(  # type: ignore # see issue #495
         transform = np.linalg.inv(affine_) @ new_affine
         # adapt to the actual rank
         transform_ = to_affine_nd(sr, transform)
+        _dtype = dtype or self.dtype or np.float32
+
+        # no resampling if it's identity transform
+        if np.allclose(transform_, np.diag(np.ones(len(transform_))), atol=1e-3):
+            output_data = data_array.copy().astype(_dtype)
+            new_affine = to_affine_nd(affine, new_affine)
+            return output_data, affine, new_affine
+
         # resample
-        _dtype = dtype or self.dtype or data_array.dtype
-        output_data = []
-        for data in data_array:
-            data_ = scipy.ndimage.affine_transform(
-                data.astype(_dtype),
-                matrix=transform_,
-                output_shape=output_shape,
-                order=self.interp_order if interp_order is None else interp_order,
-                mode=mode or self.mode,
-                cval=self.cval if cval is None else cval,
-            )
-            output_data.append(data_)
-        output_data = np.stack(output_data)
+        affine_xform = AffineTransform(
+            normalized=False,
+            mode=interp_order or self.interp_order,
+            padding_mode=mode or self.mode,
+            align_corners=True,
+            reverse_indexing=True,
+        )
+        output_data = affine_xform(
+            torch.from_numpy((data_array.astype(np.float64))[None]),  # AffineTransform requires a batch dim
+            torch.from_numpy(transform_.astype(np.float64)),
```

Comment:
A quick question: do we really need `float64` type here? It may increase the memory requirements.

Reply:
it is trying to match the precision of scipy.ndimage.affine_transform
https://github.com/scipy/scipy/blob/adc4f4f7bab120ccfab9383aba272954a0a12fb0/scipy/ndimage/interpolation.py#L449
and pytorch requires the same type for data_array and transform